agent-inject/agent/annotations.go

```diff
@@ -146,6 +147,10 @@ const (
 	// and gid) is set on the injected Vault Agent containers
 	AnnotationAgentSetSecurityContext = "vault.hashicorp.com/agent-set-security-context"
 
+	// AnnotationAgentServiceAccountTokenVolumeName is the optional name of a volume containing a
+	// service account token
+	AnnotationAgentServiceAccountTokenVolumeName = "vault.hashicorp.com/agent-service-account-token-volume-name"
```

Comment:
Please also allow defining the audience for the token with something like:
`vault.hashicorp.com/agent-service-account-token-audience`

Reply:
This PR is providing the ability to mount and use a projected token, but it's not defining the token itself. So I don't think setting the audience for the token is in scope for this PR.